src/core/include/openvino/pass/pattern/op/pattern.hpp

```diff
@@ -104,4 +104,24 @@ class OPENVINO_API Pattern : public Node {
 }  // namespace op
 }  // namespace pattern
 }  // namespace pass
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const std::function<bool(Output<Node>)>& a,
+                                                          const std::function<bool(Output<Node>)>& b);
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const pass::pattern::op::ValuePredicate& a,
+                                                          const pass::pattern::op::ValuePredicate& b);
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const pass::pattern::op::NodePredicate& a,
+                                                          const pass::pattern::op::NodePredicate& b);
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const pass::pattern::op::ValuePredicate& a,
+                                                          const pass::pattern::op::NodePredicate& b);
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const pass::pattern::op::NodePredicate& a,
+                                                          const pass::pattern::op::ValuePredicate& b);
+OPENVINO_API pass::pattern::op::ValuePredicate operator&&(const std::function<bool(Output<Node>)>& a,
```

Comment:
Is required to keep version when parameters are `std::function<bool(Output<Node>)>&` as is very similar to `ValuePredicate` alias which accepts only `const Output<Node>&`

Should be allowed to pass to predicate non const parameters?

Reply:
This was historically allowed. Since tweaking it would involve many changes in existing transformations, I simply supported the original behavior.